File: c,c++/kur1.cpp
#include<stdio.h>

int main()
{
int test,n,p,m,i,sum,pro,num;

scanf("%d",&test);

while (test--)
{
	scanf("%d %d %d",&n,&p,&m);
	sum=0;
	for(i=0;i<n;++i)
	{	
		scanf("%d",&num);
		sum+=num;
	}
	
	pro=1;

	for(i=0;i<p;++i)
	{
		pro*=sum;
		if(pro>m)
			pro=pro%m;
	}
	printf("%d\n",pro);
}

return 0;
}

File: py/kur1.py
test=raw_input()
test=int(test)

while test>0:
	line=raw_input()
	array=line.split()
	num=int(array[0])
	power=int(array[1])
	mod=int(array[2])
	s=0

	n1=raw_input()
	n=n1.split()
	s=reduce(lambda x,y:x+y , [int(i) for i in n])

	pro=pow(s,power)

	pro%=mod
	

	#for i in range(power):
	#	pro*=s
	#	if(pro>mod):
	#		pro=pro%mod


	print pro

	test=test-1
